src/Client.js

```diff
@@ -5,22 +5,79 @@ import type {
   JSONValue,
   LandRequest,
   Persona,
-  StatusEvent
+  StatusEvent,
+  Settings,
+  PullRequest,
+  BuildStatus
 } from './types';
+import Logger from './Logger';
 
 export default class Client {
   hostAdaptor: Host;
   ciAdaptor: CI;
   persona: Persona;
+  serverSettings: Settings;
 
-  constructor(host: Host, ci: CI, persona: Persona) {
+  constructor(host: Host, ci: CI, persona: Persona, settings: Settings) {
     this.hostAdaptor = host;
     this.ciAdaptor = ci;
     this.persona = persona;
+    this.serverSettings = settings;
   }
 
   async isAllowedToLand(pullRequestId: string) {
-    return await this.hostAdaptor.isAllowedToLand(pullRequestId);
+    const pullRequest: PullRequest = await this.hostAdaptor.getPullRequest(
+      pullRequestId
+    );
+    const buildStatuses = await this.hostAdaptor.getPullRequestBuildStatuses(
+      pullRequestId
+    );
+
+    const isOpen = pullRequest.state === 'OPEN';
+    const createdBy = pullRequest.author;
+    const isApproved = pullRequest.approvals.some(
```

Comment:
Later we can add number of approvals, now it's at least 1, but maybe we want at least 2 :)

Reply:
Sweet, yup